1038.py

```diff
@@ -0,0 +1,28 @@
+# Definition for a binary tree node.
+# class TreeNode:
+#     def __init__(self, val=0, left=None, right=None):
+#         self.val = val
+#         self.left = left
+#         self.right = right
+class Solution:
+    def bstToGst(self, root: TreeNode) -> TreeNode:
+        def pushleft(stack: List[TreeNode], node: TreeNode):
+            while node:
+                stack.append(node)
+                node = node.left
+
+        stack, nodes, total_sum = [], [], 0
+        pushleft(stack, root)
+
+        while stack:
+            node = stack.pop()
+            total_sum += node.val
+            nodes.append(node)
+            pushleft(stack, node.right)
+
+        for node in nodes:
```

Comment:
The beauty of this problem is that you can solve it in one-pass (modifying the value of each node, on the fly, while you traverse the tree) by slightly modifying your `pushleft` function.